### ISSUE TITLE
Cannot Import Using Typescript

### ISSUE THREAD

**Issue by ruangchupa** [senseobservationsystems/sense-error#5]
Importing using this:
`import SenseError from '@sense/error';`
returns error:
```
/Users/muhammadyusuf/Sense-Works/goalie-js/packages/sense-authentication/build/Errors.js:7
        extendStatics(d, b);
        ^


TypeError: Object prototype may only be an Object or null: undefined
```

But importing using this:
`const SenseError = require('@sense-os/error');`
Returns no error but that kind of import cannot be compiled in TS.

We need to export the module like this:
`module.exports.default = SenseError;` 
[here](https://github.com/senseobservationsystems/sense-error/blob/master/index.js#L17).

**Comment by ryubro** [comment]
I think you can do it by, either:
1. `import * as SenseError from '@sense/error';` or
2. turning the tsconfig.json option: `"allowSyntheticDefaultImports": true`

Could you try again?

**Comment by ruangchupa** [comment]
Um, I still get this error when running my code:
`TypeError: Object prototype may only be an Object or null: undefined`

When I run `tsc`, it didn't complaint anything btw.

**Comment by ryubro** [comment]
Could you add the code or branch?

**Comment by ruangchupa** [comment]
Try run the test with `npm run test` from this branch: https://github.com/senseobservationsystems/goalie-js/tree/basic-auth-functional-%234/packages/sense-authentication

**Comment by ryubro** [comment]
Hi @ruangchupa,

I could reproduce what you described.

I cannot update the repo now but mean while could you change:

- The first line In `src/Error.ts`, `import SenseError = require('@sense-os/error');`
- In `node_modules/@sense-os/error/index.d.ts`
```typescript
declare class SenseError extends Error {
    constructor(args?: { message?: string, details?: any } | string)

    message: string;
    details: any;
}

export = SenseError;
```

**Comment by ryubro** [comment]
Closed by #7
